netmaster/k8snetwork/networkpolicy.go

```diff
@@ -0,0 +1,428 @@
+package networkpolicy
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/contiv/contivmodel/client"
+	"github.com/contiv/netplugin/utils/k8sutils"
+	"github.com/rchirakk/client-go/kubernetes"
+	"github.com/rchirakk/client-go/pkg/api/v1"
+	"github.com/rchirakk/client-go/pkg/apis/extensions/v1beta1"
+	"github.com/rchirakk/client-go/pkg/watch"
+	"reflect"
+	"strings"
+	"time"
+)
+
+const defaultTenantName = "default"
+const defaultNetworkName = "net"
+const defaultSubnet = "10.1.2.0/24"
+const defaultEpgName = "ingress-group"
+const defaultPolicyName = "ingress-policy"
+const defaultRuleID = "1"
+
+type k8sContext struct {
+	k8sClientSet *kubernetes.Clientset
+	contivClient *client.ContivClient
+	isLeader     func() bool
+}
+
+var npLog *log.Entry
+
+func (k8sNet *k8sContext) handleK8sEvents() {
+	for k8sNet.isLeader() != true {
+		time.Sleep(time.Second * 10)
+	}
+
+	errCh := make(chan error)
+	for {
+		go k8sNet.watchK8sEvents(errCh)
+
+		// wait for error from api server
+		errMsg := <-errCh
+		npLog.Errorf("%s", errMsg)
+		npLog.Warnf("restarting k8s event watch")
+		time.Sleep(time.Second * 5)
+	}
+}
+
+func (k8sNet *k8sContext) createNetwork(nwName string) error {
+	npLog.Infof("create network %s", nwName)
+
+	if _, err := k8sNet.contivClient.NetworkGet(defaultTenantName, nwName); err == nil {
+		return nil
+	}
+
+	if err := k8sNet.contivClient.NetworkPost(&client.Network{
+		TenantName:  defaultTenantName,
+		NetworkName: nwName,
+		Subnet:      defaultSubnet,
+		Encap:       "vxlan",
```

Comment:
vxlan is by default like in netctl net create.
user need to create vlan network prior to applying network policy